### PR TITLE
Add the Enhancement sections in reference doc.

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.

### CODE REVIEW

**Comment by Netyyyy** [comment]
@backwind1233 please help review this.

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 1]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
+| 1
 | Max concurrent messages that the Service Bus processor client should process.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.max-concurrent-sessions
 |Integer
+|1
 | Maximum number of concurrent sessions to process at any given time.
 
-|*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.session-aware
+|*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.concurrency
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:692]: I think to put a deprecated property here is to some extent confusing. We only need introduce all the properties supported in 4.x versions. So could you move this part to the migration guide of [service bus binder properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/appendix.html#_from_azure_spring_cloud_stream_binder_servicebus_to_spring_cloud_azure_stream_binder_servicebus)? We need a table 18 to show the mapping relationship between the legacy propertis and the current one, which is similar with table 15.

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 2]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
+|Integer
+|1
+|if sessionsEnabled is true, controls the maximum number of concurrent sessions to process, else controls the max concurrent calls of service bus message handler and the size of fixed thread pool that handles user's business logic.
+
+Deprecated, replaced with maxConcurrentSessions when sessionsEnabled is true and maxConcurrentCalls when sessionsEnabled is false.
+
+|*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.session-enabled
 | Boolean
+| false
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:701]: The default value should be null.

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 3]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | Whether session is enabled.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.prefetch-count
 | Integer
+| 1
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:706]: Why is the default value 1?

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 4]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | The prefetch count of the Service Bus processor client.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.sub-queue
 | SubQueue
+| SubQueue.NONE
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:711]: Let's keep the default value format of all Enum properties in consistency, just use the string format. So here we can modify it from "SubQueue.NONE" to "none".

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 5]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | The type of the sub queue to connect to.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.max-auto-lock-renew-duration
 | Duration
+| Duration.ofMinutes(5)
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:716]: we should write the default value of each properties in the way how users will write it for a spring boot configuration file. For Duration type, you can refer to https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.conversion.durations

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 6]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | The amount of time to continue auto-renewing the lock.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.receive-mode
 | ServiceBusReceiveMode
+| PEEK_LOCK
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:721]: let's use the lower case, so change it to "peek_lock"

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 7]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | The checkpoint mode of checkpointing message. The supported modes are `MANUAL` and `RECORD`.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.max-concurrent-calls
 | Integer
+| 1
 | Max concurrent messages that the Service Bus processor client should process.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.max-concurrent-sessions
 |Integer
+|1
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:689]: why is it 1

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 8]
```diff
@@ -610,62 +665,82 @@ Configurations, Producer Properties and Advanced Producer Configurations.
...
 | Duration
+| Duration.ofMinutes(5)
 | The amount of time to continue auto-renewing the lock.
 
 |*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.receive-mode
 | ServiceBusReceiveMode
+| PEEK_LOCK
 | The receive mode of the Service Bus processor client.
 
+|*spring.cloud.stream.servicebus.bindings.<binding-name>.consumer*.enable-auto-complete
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:724]: This property is now not supported to be configured now, let's remove it

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 9]
```diff
@@ -887,6 +962,21 @@ public void producerError(Message<?> message) {
 
 [#scs-sb-headers]
 ===== Service Bus message headers
+The following table illustrates how Spring message headers are mapped to Service Bus message headers and properties. When create a message, developers can specify the header or property of a Service Bus message by below constants.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:965]: Could you check if we have had similar content already? if so we can remove it here.

File: docs/src/main/asciidoc/spring-integration-support.adoc [hunk 10]
```diff
@@ -643,6 +643,17 @@ class Demo{
 
 [#si-sb-headers]
 ===== Service Bus message headers
+
+The following table illustrates how Spring message headers are mapped to Service Bus message headers and properties. When create a message, developers can specify the header or property of a Service Bus message by below constants.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-integration-support.adoc:647]: same here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 11]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+      binders:
+        eventhub-1:
+          type: eventhubs
+          default-candidate: true
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_01}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:565]: here the property should be connection-string
image

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 12]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_02}
+                    processor:
+                      checkpoint-store:
+                        container-name: ${CHECKPOINT_CONTAINER}
+                        account-name: ${CHECKPOINT_STORAGE_ACCOUNT}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:582]: same here

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 13]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_01}
+                    processor:
+                      checkpoint-store:
+                        container-name: ${CHECKPOINT_CONTAINER}
+                        account-name: ${CHECKPOINT_STORAGE_ACCOUNT}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:569]: we should add a property of connection-string here.

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 14]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+        eventhub-2:
+          type: eventhubs
+          default-candidate: false
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_02}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:578]: same here

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 15]
```diff
@@ -892,6 +1007,109 @@ See the <<si-sb-headers, Service Bus message headers>> for the basic message hea
...
+      binders:
+        servicebus-1:
+          type: servicebus
+          default-candidate: true
+          environment:
+            spring:
+              cloud:
+                azure:
+                  servicebus:
+                    namespace: ${SERVICEBUS_NAMESPACE_01}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:1040]: same here, should be connection-string
image

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 16]
```diff
@@ -892,6 +1007,109 @@ See the <<si-sb-headers, Service Bus message headers>> for the basic message hea
...
+        servicebus-2:
+          type: servicebus
+          default-candidate: false
+          environment:
+            spring:
+              cloud:
+                azure:
+                  servicebus:
+                    namespace: ${SERVICEBUS_NAMESPACE_02}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:1049]: same here

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 17]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
 
 NOTE: When publish messages, all the above batch headers if exist will be removed from the messages to send.
 
+===== Multi Binders Support
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:534]: Let's change the title as "Multi Binder Support"

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 18]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+Step 1. To use multiBinders of EventHubs, we need to configure below properties in application.yml
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:535]: "multiBinders" should be "multiple binders". And let's add the reminder here like 
```
This sample takes connection string as example. Credentials of service priciple and MSI are also supported, users can set related properties in each binder's environment settings.
```

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 19]
```diff
@@ -892,6 +1007,109 @@ See the <<si-sb-headers, Service Bus message headers>> for the basic message hea
 
 NOTE: When setting the partiton key, the priority of message header is higher than Spring Cloud Stream property. So `spring.cloud.stream.bindings.<binding-name>.producer.partition-key-expression` will take effect only when none of the headers of `ServiceBusMessageHeaders#SESSION_ID`, `ServiceBusMessageHeaders#PARTITION_KEY`, `AzureHeaders#PARTITION_KEY` is configured.
 
+===== Multi Binders Support
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:1010]: same here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 20]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+          default-candidate: true
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_01}
+                    processor:
+                      checkpoint-store:
+                        container-name: ${CHECKPOINT_CONTAINER}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:568]: The value should be ${CHECKPOINT_CONTAINER_01}

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 21]
```diff
@@ -531,6 +531,112 @@ When the batch-consumer mode is enabled, the specific headers of batched message
...
+          default-candidate: false
+          environment:
+            spring:
+              cloud:
+                azure:
+                  eventhubs:
+                    namespace: ${EVENTHUB_NAMESPACE_02}
+                    processor:
+                      checkpoint-store:
+                        container-name: ${CHECKPOINT_CONTAINER}
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:581]: The value should be ${CHECKPOINT_CONTAINER_02}

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 22]
```diff
@@ -531,6 +531,116 @@ When the batch-consumer mode is enabled, the specific headers of batched message
 
 NOTE: When publish messages, all the above batch headers if exist will be removed from the messages to send.
 
+===== Multi Binder Support
+This sample takes connection string as example. Credentials of service priciple and MSI are also supported, users can set related properties in each binder's environment settings.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:535]: Let's add more detailed introduction here: "Connection to mutiple Event Hubs namespaces is also supported by using multi binders."

File: docs/src/main/asciidoc/spring-cloud-stream-support.adoc [hunk 23]
```diff
@@ -892,6 +1011,111 @@ See the <<si-sb-headers, Service Bus message headers>> for the basic message hea
 
 NOTE: When setting the partiton key, the priority of message header is higher than Spring Cloud Stream property. So `spring.cloud.stream.bindings.<binding-name>.producer.partition-key-expression` will take effect only when none of the headers of `ServiceBusMessageHeaders#SESSION_ID`, `ServiceBusMessageHeaders#PARTITION_KEY`, `AzureHeaders#PARTITION_KEY` is configured.
 
+===== Multi Binder Support
+This sample takes connection string as example. Credentials of service priciple and MSI are also supported, users can set related properties in each binder's environment settings.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/spring-cloud-stream-support.adoc:1015]: same here: Connection to mutiple Service Bus namespaces is also supported by using multi binders.

**Review by yiliuTo — APPROVED** [review]

**Pull request merged**